benchmarks/src/btreemap.rs

```diff
@@ -5,212 +5,141 @@ use ic_stable_structures::{storable::Blob, BTreeMap, DefaultMemoryImpl, Memory,
-    insert_blob_helper::<1024, 64>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_1024_64_v2() -> BenchResult {
-    insert_blob_helper_v2::<1024, 64>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_1024_128() -> BenchResult {
-    insert_blob_helper::<1024, 128>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_1024_128_v2() -> BenchResult {
-    insert_blob_helper_v2::<1024, 128>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_1024_256() -> BenchResult {
-    insert_blob_helper::<1024, 256>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_1024_256_v2() -> BenchResult {
-    insert_blob_helper_v2::<1024, 256>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_1024_512() -> BenchResult {
-    insert_blob_helper::<1024, 512>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_1024_512_v2() -> BenchResult {
-    insert_blob_helper_v2::<1024, 512>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_1024_512_v2_mem_manager() -> BenchResult {
-    insert_blob_helper_v2_mem_manager::<1024, 512>()
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_u64_u64() -> BenchResult {
-    let btree = BTreeMap::new_v1(DefaultMemoryImpl::default());
-    insert_helper::<u64, u64>(btree)
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_u64_u64_mem_manager() -> BenchResult {
-    let memory_manager = MemoryManager::init(DefaultMemoryImpl::default());
-    let btree = BTreeMap::new(memory_manager.get(MemoryId::new(42)));
-    insert_helper::<u64, u64>(btree)
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_u64_u64_v2() -> BenchResult {
-    let btree = BTreeMap::new(DefaultMemoryImpl::default());
-    insert_helper::<u64, u64>(btree)
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_u64_blob_8() -> BenchResult {
-    let btree = BTreeMap::new_v1(DefaultMemoryImpl::default());
-    insert_helper::<u64, Blob<8>>(btree)
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_u64_blob_8_v2() -> BenchResult {
-    let btree = BTreeMap::new(DefaultMemoryImpl::default());
-    insert_helper::<u64, Blob<8>>(btree)
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_8_u64() -> BenchResult {
-    let btree = BTreeMap::new_v1(DefaultMemoryImpl::default());
-    insert_helper::<Blob<8>, u64>(btree)
-}
-
-#[bench(raw)]
-pub fn btreemap_insert_blob_8_u64_v2() -> BenchResult {
-    let btree = BTreeMap::new(DefaultMemoryImpl::default());
-    insert_helper::<Blob<8>, u64>(btree)
+/// Helper macro to generate benchmarks.
+macro_rules! bench_tests {
+    ($( $fn_name:ident, $helper:ident, $k:expr, $v:expr );+ $(;)?) => {
+        $(
+            #[bench(raw)]
+            pub fn $fn_name() -> BenchResult {
+                $helper::<$k, $v>()
+            }
+        )+
+    };
+}
+
+type Blob4 = Blob<4>;
+type Blob8 = Blob<8>;
+type Blob16 = Blob<16>;
+type Blob32 = Blob<32>;
+type Blob64 = Blob<64>;
+type Blob128 = Blob<128>;
+type Blob256 = Blob<256>;
+type Blob512 = Blob<512>;
+type Blob1024 = Blob<1024>;
+
+// Benchmarks inserting blobs into a BTreeMap.
+bench_tests! {
```

Comment:
nit: the usual issue with the macros is that the text inside them is not auto-formatted. Below, the shorter `Blobs` seem to be not aligned with the longer blobs.

Reply:
I did that intentionally, somehow this is easier for me to read.